source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -16,6 +17,53 @@ namespace HttpFilters {
 namespace Lua {
 
 namespace {
+
+const std::string DEPRECATED_LUA_NAME = "envoy.lua";
+
+std::atomic<bool>& deprecatedNameLogged() {
+  MUTABLE_CONSTRUCT_ON_FIRST_USE(std::atomic<bool>, false);
+}
+
+bool allowDeprecatedMetadataName() {
+  if (!deprecatedNameLogged().exchange(true)) {
+    // Have not logged yet, so use the logging test.
+    return Extensions::Common::Utility::ExtensionNameUtil::allowDeprecatedExtensionName(
+        "http filter", DEPRECATED_LUA_NAME, Extensions::HttpFilters::HttpFilterNames::get().Lua);
```

Comment:
I hate to even bring this up to make this even more complicated than it already is, but technically I think someone could have an initial script that is fine, but then do an LDS reload to have a bad script which should log. In practice I think this is such an edge case that we can ignore it, but perhaps we should add a comment about this?

Reply:
I'll add a comment but I think in that case it'll log. The atomic bool path only gets invoked when we see filter metadata with "envoy.lua" as the namespace so when the script that uses the old namespace is loaded and run it'll log.

The place where it falls down the most is if someone toggles the deprecated feature flag at runtime. Then you might see only the warning version of the log message while the script is silently getting no data.

I thought about tracking whether we'd logged or not as a member variable in the filter. I think that means we'd log once per connection and I thought that still might be too much.

Reply:
Oh my bad, right, OK. Yeah I think a few more comments would be good, thank you. Also if you merge master coverage should finally be fixed I think. Thank you!

Reply:
I guess I could add another atomic to track the warn/block state we last saw and re-invoke the logging check if it changes.

Reply:
I don't think it's worth it. I would just add a comment, but up to you.